### PR TITLE
Show courses page

### PR DESCRIPTION
**Pull request by tavy88**
### Context

Show courses results page by course topic.

### Screenshots

Screen Shot 2019-07-22 at 11 25 32

Screen Shot 2019-07-21 at 14 41 03

### JIRA

https://dfedigital.atlassian.net/browse/GET-250

### How to test

1. Run CoursesService.seed
2. Visit http://localhost:3000/courses/maths and http://localhost:3000/courses/english

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -4,15 +4,16 @@ def page_title(key)
...
-    )
+    links = [
+      previous_pages.map { |title, link|
+        content_tag(:li, nil, class: 'govuk-breadcrumbs__list-item') {
+          link_to(title, link, class: 'govuk-breadcrumbs__link')
+        }
+      }
+    ]
+
+    links << content_tag(:li, current_page, class: 'govuk-breadcrumbs__list-item') if current_page
```
Comment (cwrw) [inline app/helpers/application_helper.rb]: Im not sure I understand this change, we should always have a current page, rather than a link

File: app/views/courses/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<% page_title :courses_index %>
+
+<% content_for :breadcrumb do
+    generate_breadcrumbs(
+      nil,
```
Comment (cwrw) [inline app/views/courses/index.html.erb]: I'm having a look at the prototype, and this seems like a mistake: https://get-help-to-retrain-prototype.herokuapp.com/main/release2_e2e_v1/maths_courses_near_me

Can you confirm with Steve what the wording/breadcrumb should be

File: app/views/shared/_contact_us.html.erb [hunk 3]
```diff
@@ -1,9 +1,13 @@
 <div class="govuk-grid-column-one-third">
-  <div class="govuk-related-items govuk-!-padding-bottom-4 top-border__container">
+  <div class="govuk-related-items top-border__container">
     <h2 class="govuk-heading-m"><%= t('contact_us.title') %></h2>
-    <p class="govuk-body-m govuk-!-margin-bottom-0">
-      <%= t('contact_us.body') %> <%= link_to t('contact_us.telephone'), t('contact_us.telephone_link'), class: 'govuk-!-padding-left-1 govuk-link' %>
-    </p>
-    <p class="govuk-body-m govuk-!-margin-bottom-0"><%= t('contact_us.telephone_times') %></p>
+    <ul class="govuk-list">
```
Comment (cwrw) [inline app/views/shared/_contact_us.html.erb]: why is this added to a list?

File: spec/factories/courses.rb [hunk 4]
```diff
@@ -2,12 +2,12 @@
...
     title { Faker::Educator.course_name }
     provider { Faker::Educator.university }
-    url { Faker::Internet.url }
+    url { 'http://example.com/' }
     address_line_1 { Faker::Address.street_name }
     address_line_2 { Faker::Address.street_address }
     town { Faker::Address.city }
     county { Faker::Address.state }
-    postcode { Faker::Address.postcode }
+    postcode { ['EC1A 1AA', 'EC1A 1JJ', 'EC1A 1JN', 'EC1A 2AH', 'EC1P 1AB', 'EC1P 1EW', 'EC1P 1NY'].sample }
```
Comment (cwrw) [inline spec/factories/courses.rb]: can you not set Faker to generate UK postcodes rather than this defined set?

File: spec/controllers/courses_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe CoursesController, type: :controller do
```
Comment (cwrw) [inline spec/controllers/courses_controller_spec.rb]: can we add feature specs for the courses pages as well?

File: spec/factories/courses.rb (hunk 4 above)
Comment (tavy88) [reply spec/factories/courses.rb]: I wanted to be specific so you can seed your data accordingly and get the desired distance. Will keep it like this for now.

File: app/views/shared/_contact_us.html.erb (hunk 3 above)
Comment (tavy88) [reply app/views/shared/_contact_us.html.erb]: Ask Steve :)

File: app/views/courses/index.html.erb (hunk 2 above)
Comment (tavy88) [reply app/views/courses/index.html.erb]: I followed the design :) I will have a chat with him.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (tavy88) [reply app/helpers/application_helper.rb]: Not according to the prototype?

File: spec/controllers/courses_controller_spec.rb (hunk 5 above)
Comment (tavy88) [reply spec/controllers/courses_controller_spec.rb]: What would you be testing exactly?

Comment (cwrw) [reply spec/controllers/courses_controller_spec.rb]: We should be testing  all journeys in feature tests as an integration test, I would test that the page loads with a list of Courses (can count that list)

File: app/views/courses/index.html.erb (hunk 2 above)
Comment (smoothcontract) [reply app/views/courses/index.html.erb]: Does seem like inconsistent wording 🤔

File: spec/controllers/courses_controller_spec.rb (hunk 5 above)
Comment (smoothcontract) [reply spec/controllers/courses_controller_spec.rb]: I thought we agreed to add feature specs rather than controller tests? Controller tests are basically deprecated. Could you change this to a feature spec instead please - and make sure that we see at least some course content on the page.

File: spec/factories/courses.rb (hunk 4 above)
Comment (smoothcontract) [reply spec/factories/courses.rb]: This seems okay for now.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (tavy88) [reply app/helpers/application_helper.rb]: It's not unrelated because it breaks the breadcrumbs on the prototype as it is now because we don't have a current page?

File: spec/controllers/courses_controller_spec.rb (hunk 5 above)
Comment (tavy88) [reply spec/controllers/courses_controller_spec.rb]: This was part of TDD.

File: config/locales/en.yml [hunk 6]
```diff
@@ -5,12 +5,13 @@ en-GB:
     task_list_home: 'Home: Get help to retrain'
     check_your_skills: Check your existing skills
     explore_occupations: Explore occupations
-    find_training_courses: Find and apply to training courses near you
+    find_training_courses: Training courses near you
```
Comment (tavy88) [inline config/locales/en.yml]: I didn't change this key as it will mess up our tracked page titles right now. I could change this so it looks less duplicated than find training key at line#14.

File: app/views/pages/find_training_courses.html.erb [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <% page_title :pages_find_training_courses %>
 <% content_for :breadcrumb do
     generate_breadcrumbs(
-      t('breadcrumb.find_training_courses'),
+      t('breadcrumb.find_training'),
```
Comment (cwrw) [inline app/views/pages/find_training_courses.html.erb]: :sc

File: app/views/shared/_contact_us.html.erb [hunk 8]
```diff
@@ -1,9 +1,9 @@
 <div class="govuk-grid-column-one-third">
-  <div class="govuk-related-items govuk-!-padding-bottom-4 top-border__container">
```
Comment (cwrw) [inline app/views/shared/_contact_us.html.erb]: are we sure we don't need this anymore? (padding and margins) as I recall this was needed for other pages?

Comment (tavy88) [reply app/views/shared/_contact_us.html.erb]: Hmm, I can add it back.

File: app/views/shared/_contact_us.html.erb [hunk 9]
```diff
@@ -4,6 +4,6 @@
     <p class="govuk-body-m govuk-!-margin-bottom-0">
       <%= t('contact_us.body') %> <%= link_to t('contact_us.telephone'), t('contact_us.telephone_link'), class: 'govuk-!-padding-left-1 govuk-link' %>
     </p>
-    <p class="govuk-body-m govuk-!-margin-bottom-0"><%= t('contact_us.telephone_times') %></p>
+    <p class="govuk-body-s work-hours"><%= t('contact_us.telephone_times') %></p>
```
Comment (cwrw) [inline app/views/shared/_contact_us.html.erb]: Can we keep the margin, so this is `govuk-body-s govuk-!-margin-bottom-0 muted-text` instead
